extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -20,33 +20,108 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
-	let objectName: string | undefined;
-	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
-	if (!node) {
-		// if user selects command in command palette we prompt user for information
-		quickPickStep = 'launchFromCommandPalette';
-		try {
-			// Ask binding type for promptObjectName
-			quickPickStep = 'getBindingType';
-			let selectedBinding = await azureFunctionsUtils.promptForBindingType();
-			if (!selectedBinding) {
-				return;
-			}
-			selectedBindingType = selectedBinding.type;
-			propertyBag.bindingType = selectedBindingType;
-			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
+	let isCreateNewProject: boolean = false;
+	let newFunctionFileObject: azureFunctionsUtils.IFileFunctionObject = { filePromise: undefined, watcherDisposable: undefined };
+
+	try {
+		const azureFunctionApi = await azureFunctionsUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			propertyBag.exitReason = exitReason;
+			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick)
 				.withAdditionalProperties(propertyBag).send();
+			return;
+		}
+		/**
+		 * Prompts user for azure function project path to use
+		 * If multiple found in workspace we prompt user to pick one
+		 * If one found in workspace we use that
+		 * If none found in workspace we show error message but continue with createFunction message
+		 */
+		let projectFile = await azureFunctionsUtils.getAzureFunctionProject();
+		let projectFolder: string;
+		if (!projectFile) {
+			while (true) {
+				// show warning message that user needs an azure function project to create a function
+				let projectCreate = await vscode.window.showErrorMessage(constants.azureFunctionsProjectMustBeOpened,
+					constants.createProject, constants.learnMore);
+				if (projectCreate === constants.learnMore) {
+					telemetryStep = 'learnMore';
+					void vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(constants.sqlBindingsDoc));
+					TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.learnMore)
+						.withAdditionalProperties(propertyBag).send();
+					return;
+				} else if (projectCreate === constants.createProject) {
+					telemetryStep = 'helpCreateAzureFunctionProject';
+					TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.helpCreateAzureFunctionProject)
+						.withAdditionalProperties(propertyBag).send();
+
+					isCreateNewProject = true;
+					// set projectFile to empty string so we know that the user is creating a new azure function project
+					projectFile = '';
+					let workspaceFolder = vscode.workspace.rootPath;
+					if (!workspaceFolder) {
+						// user does not have a workspace open and therefore will have to pick a folder to create the project in
```

Comment:
We should always be prompting the user where they want to save the project regardless of whether they have a workspace folder open. What's the purpose of this check?